packages/x-date-pickers/src/DateCalendar/useCalendarState.tsx

```diff
@@ -139,36 +149,25 @@ export const useCalendarState = <TDate extends unknown>({
     slideDirection: 'left',
   });
 
-  const handleChangeMonth = React.useCallback(
-    (payload: ChangeMonthPayload<TDate>) => {
-      dispatch({
-        type: 'changeMonth',
-        ...payload,
-      });
+  const slideToMonth = useEventCallback((newDate: TDate) => {
```

Comment:
Renamed to make clear that we are not updating any value, just "meta data" used to handle the visually displayed month

Reply:
If we are going this route, then maybe `onYearChange` should also be called when year changes when navigating months (not only when choosing a date in a different month—which is also in a different year)? 🤔